kedro/framework/startup.py

```diff
@@ -34,7 +36,7 @@ def _version_mismatch_error(kedro_init_version: str) -> str:
     )
 
 
-def _get_project_metadata(project_path: Union[str, Path]) -> ProjectMetadata:
```

Comment:
I hesitate a bit on whether it should accept both `str | Path`. As you can see it breaks one of our test.

On the other hand it's much cleaner to keep this path conversion happen at the beginning, i.e. `bootstrap_project`. Otherwise we are doing the same conversion over and over, and this is the only public function.